Queryable intervals

noddi opened the pull request:
Pull request for feature from #522

fjy commented on server/src/main/java/io/druid/server/http/IntervalsResource.java:
```diff
@@ -0,0 +1,202 @@
+/*
```
Let's use this header instead: 

/*
 * Licensed to Metamarkets Group Inc. (Metamarkets) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  Metamarkets licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

fjy commented on server/src/main/java/io/druid/server/http/IntervalsResource.java:
```diff
@@ -0,0 +1,202 @@
+    final Comparator<Interval> comparator = Comparators.inverse(Comparators.intervalsByStartThenEnd());
+    if (full != null) {
+      final Map<Interval, Map<String, Map<String, Object>>> retVal = Maps.newTreeMap(comparator);
+      for (DruidDataSource dataSource : this.getDataSources()) {
```
let's not explicitly use "this"

fjy commented on server/src/main/java/io/druid/server/http/IntervalsResource.java:
```diff
@@ -0,0 +1,202 @@
+        for (DataSegment dataSegment : dataSource.getSegments()) {
+          if (theInterval.contains(dataSegment.getInterval())) {
+            Map<String, Object> properties = retVal.get(dataSegment.getInterval());
+            if (properties == null) {
```
some copying and pasting here that can be broken into a function

fjy commented:
A few minor comments, but I am :+1:

fjy commented on server/src/main/java/io/druid/server/http/IntervalsResource.java:
```diff
@@ -0,0 +1,202 @@
+    return Response.ok(retVal).build();
+  }
+
+  private Set<DruidDataSource> getDataSources()
```
this is technically copy and pasted from DatasourcesResources. Might be good to find a common place for these methods.

fjy commented:
Hi @noddi, i have a few minor comments and I'm good to merge this after that.

noddi commented:
Hello @fjy 
Thanks for code review.
Changes:
- license header
- removed "this"
- moved duplicated code to setProperties method
- extracted getDatasources method to new InventoryViewUtils class

fjy commented:
@noddi Thanks for the contrib! Can you sign the CLA here: http://druid.io/community/cla.html and we'll get this merged in

xvrl commented:
:+1:

fjy commented:
Hi @noddi, have you had a chance to sign the CLA?

noddi commented:
Hello @fjy 
I'm sorry for delay, CLA signed.

xvrl commented:
Got the CLA, thanks @noddi !
